migrations/deploy.ts

```diff
@@ -0,0 +1,63 @@
+// tslint:disable:no-console
+
+import { promises as fs } from 'fs';
+import { ethers, upgrades, network } from "hardhat";
+import { getAbi } from './tools/abi';
+import { verifyProxy } from './tools/verification';
+import { getVersion } from './tools/version';
+
+
+export function getContractKeyInAbiFile(contract: string) {
+    return contract.replace(/([a-zA-Z])(?=[A-Z])/g, '$1_').toLowerCase();
+}
+
+async function main() {
+    const [ deployer,] = await ethers.getSigners();
+
+    const version = await getVersion();
+
+    // if ( ... ) {
+    //     console.log("Optional variables:");
+    //     console.log("IMA_ADDRESS");
+    //     console.log("");
+    //     console.log("For custom network do not forget to set:");
+    //     console.log("ENDPOINT - rpc endpoint");
+    //     console.log("PRIVATE_KEY - deployer private key");
+    //     console.log("GASPRICE - optional - desired gas price");
+    //     console.log("");
+    //     console.log("Usage example:");
+    //     console.log("IMA_ADDRESS=0xd2AAa00100000000000000000000000000000000 npx hardhat run migrations/deploy.ts --network custom")
+    //     process.exit(1);
+    // }
+
+    const ownerAddress = deployer.address;
+    const imaAddress = process.env.IMA_ADDRESS ? process.env.IMA_ADDRESS : ethers.constants.AddressZero;
```

Comment:
I think it would be better to have a message "Use zero address instead of MessageProxy address"